bin/evaluate-measurements.js

```diff
@@ -0,0 +1,176 @@
+import fs from 'node:fs'
+import { readFile } from 'node:fs/promises'
+import { RoundData } from '../lib/round.js'
+import { evaluate } from '../lib/evaluate.js'
+import * as SparkImpactEvaluator from '@filecoin-station/spark-impact-evaluator'
+import { fetchRoundDetails } from '../lib/spark-api.js'
+import createDebug from 'debug'
+import { Point } from '@influxdata/influxdb-client'
+import { createHash } from 'node:crypto'
+
+const { KEEP_REJECTED } = process.env
+
+const debug = createDebug('spark:bin')
+
+const [nodePath, selfPath, measurementsPath] = process.argv
+
+const USAGE = `
+Usage:
+  ${nodePath} ${selfPath} measurementsPath
+`
+
+if (!measurementsPath) {
+  console.error('Missing required argument: measurementsPath')
+  console.error(USAGE)
+  process.exit(1)
+}
+
+const keepRejected = isFlagEnabled(KEEP_REJECTED)
+
+const rounds = new Map()
+const measurementsFile = await readFile(measurementsPath, 'utf8')
+for (const line of measurementsFile.split('\n').filter(Boolean)) {
+  const { roundIndex: _roundIndex, measurement } = JSON.parse(line)
+  const roundIndex = BigInt(_roundIndex)
+  if (!rounds.has(roundIndex)) rounds.set(roundIndex, [])
+  rounds.get(roundIndex).push(measurement)
+}
+
+const EVALUATION_FILE = `evaluation-${
+  createHash('sha256').update(measurementsFile).digest('hex').substring(0, 7)
```

Comment:
Working with filenames with a (random) hash can be cumbersome, e.g. when inspecting multiple miners and relying on shell auto-completion.

Have you considered building the output filename from the input filename by adding a suffix?

E.g. `measurements-f0aaa.ndjson` -> `measurements-f0aaa-evaluated.txt`

Reply:
great idea!

https://github.com/filecoin-station/spark-evaluate/pull/401/commits/e2b5663f882b2bac2e74196540ce3bd028f07903